src/base/bittorrent/torrentinfo.cpp

```diff
@@ -305,10 +293,22 @@ int BitTorrent::TorrentInfo::fileIndex(const QString& fileName) const
     return -1;
 }
 
-void TorrentInfo::remapFiles(libtorrent::file_storage const &fileStorage)
+void TorrentInfo::stripRootFolder()
 {
-    if (!isValid()) return;
-    m_nativeInfo->remap_files(fileStorage);
+    if (filesCount() <= 1) return;
+
+    libtorrent::file_storage files = m_nativeInfo->files();
+
+    // Solution for case of renamed root folder
+    std::string testName = filePath(0).split('/').value(0).toStdString();
+    if (files.name() != testName) {
+        files.set_name(testName);
+        for (int i = 0; i < files.num_files(); ++i)
+            files.rename_file(i, files.file_path(i));
```

Comment:
Oops... I Think I remember something... And if that's not obsession, then here is the solution to more General problems than this PR has.
@qbittorrent, can someone do the following test and confirm the problem?
1. Start adding some multifile torrent via AddTorrentDialog
2. Rename its root folder (in torrent content widget)
3. Let torrent to download some files
4. Do "Set Location"

If the problem I think about really exists then qBittorrent will show you the new torrent save path but its files stay in old place.